infra/modules/providers/azure/storage-account/main.tf

```diff
@@ -0,0 +1,32 @@
+resource "azurerm_storage_account" "sa" {
+  resource_group_name      = var.resource_group_name
+  location                 = var.resource_group_location
+  name                     = lower(var.account_name)
+  account_tier             = var.performance_tier
+  account_replication_type = var.replication_type
+
+  # optional
+  account_kind              = var.kind
+  enable_https_traffic_only = var.https
+  account_encryption_source = var.encryption_source
+
+  # enrolls storage account into azure 'managed identities' authentication
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_storage_container" "sa" {
+  count                 = length(var.storage_container_names)
+  name                  = "${var.storage_container_names[count.index]}-${count.index}"
```

Comment:
I think we should keep the name that the client passed in without adding the index. This way they have full control over the naming of their app dependencies

Reply:
Makes sense! Storage account module no longer appends index count to container names. Thanks!